# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [55]:
a = 101
b = 210
c = 111
d = 402
result = ((a + b) * c) / d
print(result)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [ ]:
a=11283
b=89
a % b

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [ ]:
a=89
b=11283
a//b


0

4. Запишіть у змінну `name` ваше імʼя.

In [14]:
name = "Nelia"

5. Запишіть у змінну `surname` ваше прізвище.

In [15]:
surname = "Adzhloni"

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [43]:
name = "Nelia"
surname = "Adzhloni"
name_surname = name + " " + surname
print(name_surname)

Nelia Adzhloni


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [44]:
name_surname = "Nelia Adzhloni"
name_surname_upper = name_surname.upper()
print(name_surname_upper)

NELIA ADZHLONI


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [46]:
name_surname = "Nelia Adzhloni"
third_letter = name_surname[2]
print(third_letter)

l


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [49]:
name_surname = 'Nelia Adzhloni'
first_letter = name_surname.split()[1][0]
last_letter = name_surname.split()[1][-1]
print ((first_letter, last_letter))

('A', 'i')


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [50]:
name_surname = "Nelia Adzhloni"
print(len(name_surname.split()[1]))

8


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [51]:
city_name = "Antananarivo"
count_a = city_name.lower().count('a')
print(count_a)

4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [53]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
print(my_string.startswith("The"))
print(my_string.endswith("dog!"))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [54]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
result = "_".join(my_string.split())
print(result)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`